src/editor/controls/GitHubRepositoryControl.tsx

```diff
@@ -0,0 +1,119 @@
+import {BaseControl, Button, TextControl,} from '@wordpress/components';
+import {useEffect, useRef} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+
+interface GitHubRepositoryControlProps {
+    value: string;
+    onChange: (value: string) => void;
+}
+
+export const GitHubRepositoryControl = ({value, onChange, onCodeFetched}: GitHubRepositoryControlProps) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+        if (isValidUrl(value)) {
+            fetchFile(value).then((code) => {
+                if (value.indexOf("#L")) {
+                    onCodeFetched({ code, lineNumbers: extractLineNumbers(value) });
+                } else {
+                    onCodeFetched({ code });
+                }
+            });
+
+            inputRef.current
+                ?.querySelector('input')
+                ?.classList.remove('cbp-input-error');
+        } else {
+            inputRef.current
+                ?.querySelector('input')
+                ?.classList.add('cbp-input-error');
+        }
+    }, [value]);
+    return (
+        <BaseControl id="code-block-pro-show-highlighting">
```

Comment:
this just got copied over. I believe these need to be unique and correspont with an input inside (the TextControl) but im not 100% sure actually.